legacy/messaging/src/main/java/org/jboss/as/messaging/MigrateOperation.java

```diff
@@ -494,12 +495,16 @@ private boolean connectionFactoryIsUsingInVMConnectors(OperationContext context,
     private void migrateDiscoveryGroup(ModelNode newAddOp, List<String> warnings) {
         // These attributes are not present in the messaging-activemq subsystem.
         // Instead a socket-binding must be used to configure the broadcast-group.
+        boolean mandatorySocketBinding = false;
         for (String property : asList(LOCAL_BIND_ADDRESS.getName(), GROUP_ADDRESS.getName(), GROUP_PORT.getName())) {
             if (newAddOp.has(property)) {
                 newAddOp.remove(property);
                 warnings.add(ROOT_LOGGER.couldNotMigrateDiscoveryGroupAttribute(property, pathAddress(newAddOp.get(OP_ADDR))));
+                mandatorySocketBinding = true;
             }
         }
+        if (mandatorySocketBinding)
+            newAddOp.get(SOCKET_BINDING.getName()).set("group-x-binding");
```

Comment:
Thanks, I see your point now, but do you mean an error instead of warn? Since one of the broadcast-group or discovery-groups becomes invalid, it then fails the migration operation and stops after the first failure.
I think I will complete the subsystem_migration.xml with missing alternative attributes to keep the current tests for successful migration with different warnings, then a new test with files like subsystem_migration_broadcast-groups_error.xml and subsystem_migration_discovery-groups_error.xml to expect migration error to happen in case of invalid attributes. wdyt?